### PR TITLE
Switching SSW website to use JotForms

### LINKED ISSUE #1619: Forms - switch to Jotform
Current stack:

- Custom booking form on website, different one on people, different one on rules, different one for SSWTV, etc
- Need to store keys for PowerAutomate, Recaptcha

Proposed solution:

- Move form to Jotform <https://www.jotform.com/pricing/>
- Have 1 definition of the booking form (reused on all sites)
- Then use Jotform for UG bookings

```[tasklist]
### Tasks
- [x] Get purchase please approved
- [x] Migrate booking form to JotForm
- [x] Create a new Power Automate flow for submission
- [x] Disable Weekly - Test lead email PA Flow
- [x] ~Remove unused packages/secrets from app~ Once people and rules are not using current api then we will delete it
- [x] Reply done to "Re: Consulting enquiry - Metro Finance - Chevon Abbaticchio"
```

### PR DESCRIPTION
**Pull request by amankumarrr**
- Affected routes:`/*`
- Fixed #1619 

- [ ] Include done video or screenshots

image

**Figure: New Booking form from the JotForm**

### CODE REVIEW

File: .github/.env [hunk 1]
```diff
@@ -8,3 +8,4 @@ KEY_VAULT=kv-sswwebsite-9eb3
 APP_INSIGHT_PROD=appInsight-sswwebsite-9eb3
 APP_INSIGHT_DEV=appInsight-sswwebsite-dev-9eb3
 NEXT_PUBLIC_CHATBASE_BOT_ID=gnbZXuG_5tgI_R3yzKcOn
+NEXT_PUBLIC_JOTFORM_ID=233468468973070
```
Comment (wicksipedia) [inline .github/.env]: ideally, this should be updatable in tina

Comment (amankumarrr) [reply .github/.env]: Fixed

File: .tina/collections/global.tsx [hunk 2]
```diff
@@ -239,5 +239,71 @@ export const globalSchema: Collection = {
         },
       ],
     },
+    {
+      type: "object",
+      name: "jotForm",
+      label: "JotForm",
```
Comment (wicksipedia) [inline .tina/collections/global.tsx]: is this a component I can drop on any page?

File: components/bookingButton/bookingButton.tsx [hunk 3]
```diff
@@ -38,39 +28,23 @@ export const BookingButton = ({ data }) => {
...
-        </h2>
-      )}
-      <Popup isVisible={isVisible} onClose={setIsVisible}>
-        {isVisible && (
-          <BookingForm
-            onClose={setIsVisible}
-            showSuccessToast={showSuccessToast}
-          />
+    <>
+      <JotFormScript />
```
Comment (wicksipedia) [inline components/bookingButton/bookingButton.tsx]: I think this shouldn't be hardcoded here - makes it hard to reuse on other pages

File: components/bookingButton/bookingButton.tsx [hunk 4]
```diff
@@ -38,39 +28,23 @@ export const BookingButton = ({ data }) => {
...
-
-  const showSuccessToast = () => {
-    toast.success(
-      <div id="success-toaster" className="text-left">
-        Form submitted. We&apos;ll be in contact as soon as possible.
-      </div>
-    );
-  };
+  const jotFormClass =
+    buttonClass ?? "mt-14" + " " + `lightbox-${layoutData.jotForm.id}`; // lightbox-id class is a trigger point for the JotForm IFrame
```
Comment (wicksipedia) [inline components/bookingButton/bookingButton.tsx]:
```suggestion
    buttonClass ?? `mt-14 lightbox-${layoutData.jotForm.id}`; // lightbox-id class is a trigger point for the JotForm IFrame
```

File: components/bookingButton/bookingButton.tsx [hunk 5]
```diff
@@ -38,39 +28,23 @@ export const BookingButton = ({ data }) => {
...
-          <BookingForm
-            onClose={setIsVisible}
-            showSuccessToast={showSuccessToast}
-          />
+    <>
+      <JotFormScript />
+      <div
+        className={twMerge("flex w-full flex-col items-center", containerClass)}
+      >
+        <UtilityButton className={jotFormClass} buttonText={buttonText} />
```
Comment (wicksipedia) [inline components/bookingButton/bookingButton.tsx]: if it is just the class name that wires up the button to the class - better to have a jotform button component that abstracts that implementation detail away

File: .tina/collections/global.tsx (hunk 2 above)
Comment (amankumarrr) [reply .tina/collections/global.tsx]: Fixed - added a component that can be reused

File: components/bookingButton/bookingButton.tsx (hunk 4 above)
Comment (amankumarrr) [reply components/bookingButton/bookingButton.tsx]: fixed

File: components/bookingButton/bookingButton.tsx (hunk 5 above)
Comment (amankumarrr) [reply components/bookingButton/bookingButton.tsx]: Fixed - added a component that can be reused

File: components/bookingButton/bookingButton.tsx (hunk 3 above)
Comment (amankumarrr) [reply components/bookingButton/bookingButton.tsx]: Fixed - added a component that can be reused

File: components/blocks/jotFormEmbed.tsx [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+      windowTitle: '${jotForm?.formTitle}',
+      backgroundColor: '${jotForm?.backgroundColor}',
+      fontColor: '${jotForm?.fontColor}',
+      type: '${jotForm?.formType}',
+      height: ${jotForm?.height},
+      width: ${jotForm?.width},
+      openOnLoad: false
+    });
+
+    var ifr = document.getElementById("lightbox-233468468973070");
```
Comment (wicksipedia) [inline components/blocks/jotFormEmbed.tsx]: is this jotform id meant to be interpolated and not hardcoded?

Comment (amankumarrr) [reply components/blocks/jotFormEmbed.tsx]: I missed that, Fixed it

**Review by wicksipedia — APPROVED** [review]

**Pull request merged**